src/libraries/System.IO.Ports/src/System/IO/Ports/SerialPort.Linux.cs

```diff
@@ -33,11 +33,13 @@ public static string[] GetPortNames()
                     // checking for that for non-ttyS entries is incorrect as some uart
                     // devices are incorrectly filtered out
                     bool isTtyS = entry.Name.StartsWith("ttyS");
+                    bool isTtyGS = !isTtyS && entry.Name.StartsWith("ttyGS");
```

Comment:
Might be better to have a helper function to match not only this one (and also use in the path where `/dev` is enumerated) against a regex.  For instance, `tty[A-Za-z]+[0-9]+` should match all non-terminal TTY devices (which would match `tty[0-9]+`).  We'll never be able to keep up with new device names.

Reply:
I'd recommend creating an issue about it with suggested solution. So far we haven't been able to find any way to properly detect it other than combination of name and probing for some files. Also I don't believe new device types show up that often so I don't find it as super high priority. Note that this is a convenience API and if device is in fact TTY then passing it to constructor will work even if it's not listed by GetPortNames